settings.gradle

```diff
@@ -1 +1,5 @@
+plugins {
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.7.0"
```

Comment:
Is this in use?

Reply:
To some degree, yes. Once the environment meets the minimum requirements (such as JDK >= 17 for Android Gradle Plugin 8.2), this auto-provisioning plugin automatically downloads the JDK version specified by the Java toolchain setting. I followed the documentation from [here](https://developer.android.com/build/jdks#toolchain) and [here](https://docs.gradle.org/current/userguide/toolchains.html) and tested its behaviour by changing this setting:

```groovy
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(17)
    }
}
```

e.g. to `JavaLanguageVersion.of(20)`. It automatically downloaded and used JDK 20 for me, so it was quite convenient. I also confirmed it was actually used for compilation by running `javap -verbose` on one of the classes from the AAR. It gave me `major version: 64` - matching JDK 20.

Again, the environment must meet the minimum requirements first. Otherwise, this mechanism doesn't work. That's why we still need things like `jitpack.yml` (JitPack runners have only JDK 8 by default).